src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +153,97 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        private void DetectScriptLocation()
+        {
+            if (_scriptDetectionFailed || AutoConfigUrl != null)
```

Comment:
DetectScriptLocation is only called from the ctor, at the very beginning.  How could _scriptDetectionFailed be true?  You can make this a Debug.Assert(!_scriptDetectionFailed) if you want to validate that, but I don't think that's even necessary.

Reply:
Same comment for AutoConfigUrl... how could it be non-null at this point?

Reply:
Good catch, while I was writing this function in the first place, the design was that it could be call multiple times. Adding cache like this will save time calling into it.

Later after refactoring the code and moving it to the ctor, it only serves the purpose that to help us determine if a proxy is used or not. The logic here is not useful at all. Will remove it.